collectors/prometheus.go

```diff
@@ -94,11 +96,22 @@ func NewPrometheusExporter(cfg *PrometheusConfig, lnd *lndclient.LndServices,
 
 	htlcMonitor := newHtlcMonitor(lnd.Router, errChan)
 
+	// Setup signalling so SIGTERM || SIGINT will stop the cache goroutine
+	// from ChannelsCollector
+	quit := make(chan struct{})
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
We already create a `signal.Intercept()` instance in `lndmon.go` which handles this. So I think we should create a quit channel there that is passed into the `NewPrometheusExporter` function and is closed here: https://github.com/lightninglabs/lndmon/blob/a03710fbc14151aa6bbda13ec989f5385a731185/lndmon.go#L86